internal/controllers/addon/handle_installed_condition.go

```diff
@@ -54,11 +65,51 @@ func (r *olmReconciler) deleteConfigMapPresent(ctx context.Context, addon *addon
 		Namespace: addonNamespace,
 	}, deleteConfigMap)
 	if err != nil {
-		if errors.IsNotFound(err) {
+		if k8serrors.IsNotFound(err) {
 			return false, nil
 		}
 		return false, err
 	}
 	_, found := deleteConfigMap.Labels[deleteConfigMapLabel]
 	return found, nil
 }
+
+// handleInstallAck handles installation acknowledgement from
+// the addon instance if specified in the addon CR
+func (r *olmReconciler) handleInstallAck(ctx context.Context, addon *addonsv1alpha1.Addon) (requeueResult, error) {
+	installed, err := r.isAddonInstanceInstalled(ctx, addon)
+	if err != nil {
+		return resultNil, err
+	}
+
+	if !installed {
+		return resultRetry, nil
+	}
+
+	reportInstalledCondition(addon)
+	return resultNil, nil
+}
+
+// isAddonInstanceInstalled returns if the corresponding addon instance has installed=true status condition
+func (r *olmReconciler) isAddonInstanceInstalled(ctx context.Context, addon *addonsv1alpha1.Addon) (bool, error) {
+	addonInstance := &addonsv1alpha1.AddonInstance{}
+	instanceKey := types.NamespacedName{
+		Name:      addonsv1alpha1.DefaultAddonInstanceName,
+		Namespace: GetCommonInstallOptions(addon).Namespace,
+	}
+
+	if err := r.client.Get(ctx, instanceKey, addonInstance); err != nil {
+		return false, err
+	}
+
+	if installedCond := meta.FindStatusCondition(
+		addonInstance.Status.Conditions,
+		addonsv1alpha1.AddonInstanceConditionInstalled.String(),
+	); installedCond == nil {
```

Comment:
Sorry i missed this too, Don't we want to report the addon as pending when `AddonInstanceConditionInstalled`== false.